inc/storefront-template-functions.php

```diff
@@ -158,19 +158,28 @@ function storefront_header_widget_region() {
 	 * @since  1.0.0
 	 * @return void
 	 */
-	function storefront_site_branding() {
-		if ( function_exists( 'the_custom_logo' ) && has_custom_logo() ) {
-			the_custom_logo();
-		} elseif ( function_exists( 'jetpack_has_site_logo' ) && jetpack_has_site_logo() ) {
-			jetpack_the_site_logo();
-		} else { ?>
-			<div class="site-branding">
-				<h1 class="site-title"><a href="<?php echo esc_url( home_url( '/' ) ); ?>" rel="home"><?php bloginfo( 'name' ); ?></a></h1>
-				<?php if ( '' != get_bloginfo( 'description' ) ) { ?>
+	function storefront_site_branding() { ?>
+    <div class="site-branding">
+      <?php
+      if ( function_exists( 'the_custom_logo' ) && has_custom_logo() ) {
+        $logo = get_custom_logo();
+
+        echo $logo = is_home() ? '<h1>' . $logo . '</h1>' : $logo;
```

Comment:
Hi @jameskoster 

I see that you changed this to `is_front_page()` in the last update.

The problem with this new approach becomes obvious if we use a "static" one as front page like the WooCommerce Shop page... We don't want the logo as heading on this page but only the page title at the top level.

To make short, wrapping the logo inside `<h1>` is good only on home page (Posts page) since it doesn't have a heading itself.